### PR TITLE
CDAP-3746 and CDAP-3801 : Verify the cassandra.addresses and enhance …

### PR DESCRIPTION
**Pull request by gokulavasan**
…tool tip for schema property

And make rpcPort property nullable (default value is 9160) 

Build : http://builds.cask.co/browse/CPG-RCP0

### CODE REVIEW

File: cassandra-plugins/src/main/java/co/cask/plugin/etl/batch/sink/BatchCassandraSink.java [hunk 1]
```diff
@@ -131,9 +132,10 @@ private ByteBuffer encodeObject(Object object, Schema schema) throws IOException
     private String partitioner;
 
     @Name(Cassandra.PORT)
+    @Nullable
     @Description("The rpc port for Cassandra. For example, 9160. " +
```
Comment (awholegunch) [inline cassandra-plugins/src/main/java/co/cask/plugin/etl/batch/sink/BatchCassandraSink.java]: The RPC port...

**Comment by awholegunch** [comment]
I've made edits. Please check. Otherwise, LGTM.

File: cassandra-plugins/src/main/java/co/cask/plugin/etl/realtime/RealtimeCassandraSink.java [hunk 2]
```diff
@@ -89,6 +89,17 @@ public void configurePipeline(PipelineConfigurer pipelineConfigurer) {
...
                                   " and password: " + config.password);
+    Preconditions.checkArgument(!parseAddresses(config.addresses).isEmpty(),
+                                "At least one pair of IP and port should be provided.",
+                                config.addresses, ADDRESSES_DESCRIPTION);
+  }
+
+  private List<InetSocketAddress> parseAddresses(String addressString) {
+    List<InetSocketAddress> addresses = new ArrayList<>();
+    for (String address : addressString.split(",")) {
+      addresses.add(new InetSocketAddress(address.split(":")[0], Integer.valueOf(address.split(":")[1])));
```
Comment (albertshau) [inline cassandra-plugins/src/main/java/co/cask/plugin/etl/realtime/RealtimeCassandraSink.java]: split it once and do some verification. For example, check there are exactly 2 parts and make sure we throw useful error messages if the port is not a number

**Comment by albertshau** [comment]
1 comment about error handling

**Comment by gokulavasan** [comment]
@awholegunch Thanks for the fixes JJ.

**Comment by gokulavasan** [comment]
@albertshau Addressed comment. Please take a look again.

**Comment by albertshau** [comment]
:+1:

**Pull request merged**